packages/expo-modules-core/ios/Core/SharedObjects/SharedObject.swift

```diff
@@ -17,8 +17,37 @@ open class SharedObject: AnySharedObject {
    */
   public internal(set) var sharedObjectId: SharedObjectId = 0
 
+  /**
+   An app context for which the shared object was created.
+   */
+  public internal(set) weak var appContext: AppContext?
+
   /**
    The default public initializer of the shared object.
    */
   public init() {}
+
+  /**
+   Returns the JavaScript shared object associated with the native shared object.
+   */
+  public func getJavaScriptObject() -> JavaScriptObject? {
+    return appContext?.sharedObjectRegistry.toJavaScriptObject(self)
+  }
```

Comment:
This was removed in https://github.com/expo/expo/pull/27032, but we need this again 😄